### PR TITLE
fix: Move the code from bloomstack_core to erpnext for packing slip w…

### PR DESCRIPTION
**Pull request by shruti2323**
Fix Description: Moved the code from bloomstack_core to erpnext along with its child doc.
Asana Link: https://app.asana.com/0/1200073273462988/1199942760636622

**PR of bloomstack core fix: https://github.com/Bloomstack/bloomstack_core/pull/610**

### CODE REVIEW

File: erpnext/stock/doctype/packing_slip/packing_slip.py [hunk 1]
```diff
@@ -30,6 +31,9 @@ def validate(self):
 		validate_uom_is_integer(self, "stock_uom", "qty")
 		validate_uom_is_integer(self, "weight_uom", "net_weight")
 
+	def on_submit(self):
```
Comment (shruti2323) [inline erpnext/stock/doctype/packing_slip/packing_slip.py:34]: Had added this code from bloomstack_core to erpnext where under hooks.py files this part of code runs on event : "on submit". Similarly made the functionality to be run on "on submit" event.

File: erpnext/stock/doctype/packing_slip_item/packing_slip_item.json [hunk 2]
```diff
@@ -12,6 +12,10 @@
   "batch_no",
   "desc_section",
   "description",
+  "warehouse_details_section",
```
Comment (shruti2323) [inline erpnext/stock/doctype/packing_slip_item/packing_slip_item.json:15]: Kindly refer the image.
packing_slip_item_fields

https://user-images.githubusercontent.com/80836439/123172346-a5676180-d49a-11eb-838f-2a3443285ed8.mp4

File: erpnext/stock/doctype/packing_slip/packing_slip.json [hunk 3]
```diff
@@ -1,264 +1,290 @@
 {
```
Comment (shruti2323) [inline erpnext/stock/doctype/packing_slip/packing_slip.json:1]: Kindly refer below images:


https://user-images.githubusercontent.com/80836439/123172540-e8c1d000-d49a-11eb-9e50-ae92e4572419.mp4

Packing_slip_Fields
Packing_slip_doctype_fields_before
Packing_slip_custom_fields

File: erpnext/stock/doctype/packing_slip/packing_slip.json [hunk 4]
```diff
@@ -1,264 +1,290 @@
...
+   "fieldtype": "Link",
+   "ignore_user_permissions": 1,
+   "label": "Amended From",
+   "no_copy": 1,
+   "options": "Packing Slip",
+   "print_hide": 1,
+   "read_only": 1
+  },
+  {
+   "fieldname": "source_warehouse",
```
Comment (shruti2323) [inline erpnext/stock/doctype/packing_slip/packing_slip.json:184]: Refer below image: 
Packing_slip_Fields
Packing_slip_doctype_fields_before
Packing_slip_custom_fields

https://user-images.githubusercontent.com/80836439/123172745-33dbe300-d49b-11eb-8ad5-0ab1609b66d0.mp4

**Review by hrwX — APPROVED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**